src/cube_neon.h

```diff
@@ -1,6 +1,423 @@
+	{
+		piece = corners[i];
+		orien = ((piece << 1) | (piece >> 1)) & _cobits2;
+		corner_result[piece & _pbits] = i | orien;
+	}
+
+	// Copy the results back to the NEON vectors
+	ret.edge = vld1q_u8(edge_result);
+	ret.corner = vld1q_u8(corner_result);
+
+	return ret;
+}
+
+_static_inline int64_t coord_co(cube_t c)
+{
+	// Temp array to store the NEON vector
+	uint8_t mem[16];
+	vst1q_u8(mem, c.corner);
+
+	int i, p;
+	int64_t ret;
+
+	for (ret = 0, i = 0, p = 1; i < 7; i++, p *= 3)
+		ret += p * (mem[i] >> _coshift);
+
+	return ret;
+}
+
+_static_inline int64_t
+coord_csep(cube_t c)
+{
+	// Temp array to store the NEON vector
+	uint8_t mem[16];
+	vst1q_u8(mem, c.corner);
+
+	int64_t ret = 0;
+	int i, p;
+	for (ret = 0, i = 0, p = 1; i < 7; i++, p *= 2)
+		ret += p * ((mem[i] & _csepbit) >> 2);
+
+	return ret;
+	return 0;
+}
+
+_static_inline int64_t
+coord_cocsep(cube_t c)
+{
+	return (coord_co(c) << 7) + coord_csep(c);
+}
+
+_static_inline int64_t coord_eo(cube_t c)
+{
+	int64_t ret = 0;
+	int64_t p = 1;
+
+	// Temp array to store the NEON vector
+	uint8_t mem[16];
+	vst1q_u8(mem, c.edge);
+
+	for (int i = 1; i < 12; i++, p *= 2)
+	{
+		ret += p * (mem[i] >> _eoshift);
+	}
+
+	return ret;
+}
+
+_static_inline int64_t coord_esep(cube_t c)
+{
+	int64_t i, j, jj, k, l, ret1, ret2, bit1, bit2, is1;
+
+	// Temp array to store the NEON vector
+	uint8_t mem[16];
+	vst1q_u8(mem, c.edge);
+
+	for (i = 0, j = 0, k = 4, l = 4, ret1 = 0, ret2 = 0; i < 12; i++)
+	{
+		bit1 = (mem[i] & _esepbit1) >> 2;
+		bit2 = (mem[i] & _esepbit2) >> 3;
+		is1 = (1 - bit2) * bit1;
+
+		ret1 += bit2 * binomial[11 - i][k];
+		k -= bit2;
+
+		jj = j < 8;
+		ret2 += jj * is1 * binomial[7 - (j * jj)][l];
+		l -= is1;
+		j += (1 - bit2);
+	}
+
+	return ret1 * 70 + ret2;
+}
+
+_static_inline void
+copy_corners(cube_t *dst, cube_t src)
+{
+	dst->corner = src.corner;
+}
+
+_static_inline void
+copy_edges(cube_t *dst, cube_t src)
+{
+	dst->edge = src.edge;
+}
+
+_static_inline void
+set_eo(cube_t *cube, int64_t eo)
+{
+	// Temp array to store the NEON vector
+	uint8_t mem[16];
+	vst1q_u8(mem, cube->edge);
+	uint8_t i, sum, flip;
+
+	for (sum = 0, i = 1; i < 12; i++, eo >>= 1)
+	{
+		flip = eo % 2;
+		sum += flip;
+		mem[i] = (mem[i] & ~_eobit) | (_eobit * flip);
+	}
+	mem[0] = (mem[0] & ~_eobit) | (_eobit * (sum % 2));
+
+	// Copy the results back to the NEON vector
+	cube->edge = vld1q_u8(mem);
+	return;
+}
+
+_static_inline cube_t
+invcoord_esep(int64_t esep)
+{
+	cube_t ret;
+	int64_t bit1, bit2, i, j, jj, k, l, s, v, w, is1, set1, set2;
+	uint8_t slice[3] = {0};
+
+	ret = solved;
+	uint8_t mem[16];
+	set1 = esep % 70;
+	set2 = esep / 70;
+
+	for (i = 0, j = 0, k = 4, l = 4; i < 12; i++)
+	{
+		v = binomial[11 - i][k];
+		jj = j < 8;
+		w = jj * binomial[7 - (j * jj)][l];
+		bit2 = set2 >= v;
+		bit1 = set1 >= w;
+		is1 = (1 - bit2) * bit1;
+
+		set2 -= bit2 * v;
+		k -= bit2;
+		set1 -= is1 * w;
+		l -= is1;
+		j += (1 - bit2);
+		s = 2 * bit2 + (1 - bit2) * bit1;
+
```

Comment:
Mi sa che questi calcoli li abbiamo scritti tre volte uguali, magari li sistemo in utils.h o qualcosa del genere (faccio io).

Reply:
Per questa funzione ho seguito ciecamente l'implementazione delle altre versioni, dato che non ci sono ottimizzazioni presenti in AVX2.